redis/client.go

```diff
@@ -220,6 +220,13 @@ func (c *Client) XReadUntilResult(ctx context.Context, a *redis.XReadArgs) ([]re
 				continue
 			}
 
+			var timeout interface {
+				Timeout() bool
+			}
+			if errors.As(err, &timeout) && timeout.Timeout() && !errors.Is(err, context.DeadlineExceeded) {
+				continue
+			}
```

Comment:
1. We already have a function with the purpose to test if an error is considered retryable, why isn't it used here, is there a reason why some of the errors it consideres retryable shouldn't be retried here? https://github.com/Icinga/icinga-go-library/blob/420fbff2ee6fc4ff2bc53df3fe9d4f1ffb64570c/retry/retry.go#L135-L201
1. Explicitly checking for `context.DeadlineExceeded` feels like a very narrow fix for https://github.com/Icinga/icinga-go-library/pull/23#discussion_r1705689655. What about having `ctx.Err() == nil` as an additional condition for a retry? That sounds like a sane check in general, avoids the problem as well, and doesn't rely on any context error internals.

So maybe just do something like this instead (I can't mark it as a suggestion as it also subsumes the previous `if` which is one line out of the diff context):

```go
			if (errors.Is(err, redis.Nil) || retry.Retryable(err)) && ctx.Err() == nil {
				continue
			}
```